docs/auth/index.md

```diff
@@ -278,3 +289,16 @@ createApiFactory({
     ),
 });
 ```
+
+\_**Warning:** You will need to remove the default `ScmAuth.createDefaultApiFactory();` with this approach.
+
+3.  Finally you also need to add and configure another GitHub provider to the `auth-backend` using the provider ID, which in this example is `ghe`:
```

Comment:
The text here is written in a pretty prescriptive way; to me, reading it from top to bottom, it loses its generality (of telling how to make ANY ScmAuthApi) and starts describing the ghe specific case and its nuances pretty closely.

To be clear, it's perfectly fine to make an `ScmAuthApi` that has only the default github provider and the default bitbucket provider in it and nothing else. So, adding a new auth API ref is optional, and adding a provider factory in the backend is also optional. It depends on the circumstances, and the latter is kinda detached from the act of making the `ScmAuthApi` specifically.

Could the text be reworded a bit to clarify this? Maybe the re could be an initial paragraph that only says how to make a custom instance of the scmAuthApiRef (including removing the default one). And then a follow-up paragraph that says that yeah you can also add your entirely own auth refs and inject them into the scm auth api, and you do that by ... etc

Reply:
@freben I have updated the copy, but I actually do not know when you need to add a n entry to `providerFactories` and when you do not so any help on that would be great.